app/forms/Transport.scala

```diff
@@ -42,15 +42,33 @@ object Transport {
 
   import ModesOfTransport._
 
-  val allowedModeOfTransport = Seq(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
+  val allowedModeOfTransport =
+    Seq(Sea, Rail, Road, Air, PostalOrMail, FixedInstallations, InlandWaterway, Other)
 
   val mapping = Forms.mapping(
     "modeOfTransport" -> requiredRadio("transport.modeOfTransport.empty")
       .verifying("transport.modeOfTransport.error", isContainedIn(allowedModeOfTransport)),
     "nationality" -> text()
       .verifying("transport.nationality.empty", nonEmpty)
-      .verifying("transport.nationality.error", isEmpty or isValidCountryCode)
+      .verifying("transport.nationality.error", isEmpty or isValidCountryCode),
+    "transportId" -> text()
+      .verifying("transport.transportId.empty", nonEmpty)
+      .verifying("transport.transportId.error", isEmpty or (noLongerThan(35) and isAlphanumeric))
   )(Transport.apply)(Transport.unapply)
 
-  def form(): Form[Transport] = Form(mapping)
+  def form: Form[Transport] =
+    Form(mapping)
+
+  def messageKey(mode: String) =
+    s"transport.modeOfTransport.${mode match {
+      case ModesOfTransport.Sea                => "sea"
+      case ModesOfTransport.Rail               => "rail"
+      case ModesOfTransport.Road               => "road"
+      case ModesOfTransport.Air                => "air"
+      case ModesOfTransport.PostalOrMail       => "postalOrMail"
+      case ModesOfTransport.FixedInstallations => "fixed"
+      case ModesOfTransport.InlandWaterway     => "inlandWaterway"
+      case ModesOfTransport.Other              => "other"
+      case _                                   => "unknown"
```

Comment:
You're adding unknown, build whole message key `transport.modeOfTransport.unknown`

We don't have key like this and you didn't add it.

Reply:
'unknown' should never happen but it it did I think it would be preferable to the app blowing up with match exception.  I'll add a message key for 'unknown' as well